File: app/src/main/java/com/mt/mt166demo/ConnDialog.java
package com.mt.mt166demo;

import android.app.Dialog;
import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.Spinner;
import android.widget.SpinnerAdapter;
import java.util.Vector;

/* loaded from: classes2.dex */
public class ConnDialog extends Dialog {
    OnCloseListener mCloseListener;
    Context mContext;
    Spinner mSpinBaudRates;
    Spinner mSpinPorts;

    public interface OnCloseListener {
        void onClose(boolean z);
    }

    public ConnDialog(Context context) {
        super(context);
        this.mCloseListener = null;
        this.mContext = context;
    }

    public void setCloseListener(OnCloseListener listener) {
        this.mCloseListener = listener;
    }

    @Override // android.app.Dialog
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_conn);
        setTitle(MainActivity.instance.getString(R.string.deviceconnect));
        setCancelable(false);
        this.mSpinPorts = (Spinner) findViewById(R.id.spinPorts);
        this.mSpinBaudRates = (Spinner) findViewById(R.id.spinBaudRates);
        SerialPortFinder finder = new SerialPortFinder();
        Vector<SerialPortItem> ports = finder.getAllPorts();
        if (ports.size() != 0) {
            ArrayAdapter<SerialPortItem> adapter1 = new ArrayAdapter<>(this.mContext, R.layout.support_simple_spinner_dropdown_item, ports);
            this.mSpinPorts.setAdapter((SpinnerAdapter) adapter1);
        } else {
            Vector<SerialPortItem> ports2 = new Vector<>();
            ports2.add(new SerialPortItem("COM0", "/dev/ttyS0"));
            ports2.add(new SerialPortItem("COM1", "/dev/ttyS1"));
            ports2.add(new SerialPortItem("COM2", "/dev/ttyS2"));
            ports2.add(new SerialPortItem("COM3", "/dev/ttyS3"));
            ports2.add(new SerialPortItem("COM4", "/dev/ttyS4"));
            ArrayAdapter<SerialPortItem> adapter12 = new ArrayAdapter<>(this.mContext, R.layout.support_simple_spinner_dropdown_item, ports2);
            this.mSpinPorts.setAdapter((SpinnerAdapter) adapter12);
        }
        Vector<String> baudRates = new Vector<>();
        baudRates.add("9600");
        baudRates.add("19200");
        baudRates.add("38400");
        baudRates.add("57600");
        baudRates.add("115200");
        ArrayAdapter<String> adapter2 = new ArrayAdapter<>(this.mContext, R.layout.support_simple_spinner_dropdown_item, baudRates);
        this.mSpinBaudRates.setAdapter((SpinnerAdapter) adapter2);
        SharedPreferences sp = this.mContext.getSharedPreferences("conn_pref", 0);
        String lastPortName = sp.getString("PortName", "");
        int i = 0;
        while (true) {
            if (i >= this.mSpinPorts.getCount()) {
                break;
            }
            if (!((SerialPortItem) this.mSpinPorts.getItemAtPosition(i)).getName().equalsIgnoreCase(lastPortName)) {
                i++;
            } else {
                this.mSpinPorts.setSelection(i);
                break;
            }
        }
        int i2 = 0;
        while (true) {
            if (i2 >= this.mSpinBaudRates.getCount()) {
                break;
            }
            if (Integer.parseInt((String) this.mSpinBaudRates.getItemAtPosition(i2)) != 9600) {
                i2++;
            } else {
                this.mSpinBaudRates.setSelection(i2);
                break;
            }
        }
        Button OkButton = (Button) findViewById(R.id.btnOK);
        OkButton.setOnClickListener(new View.OnClickListener() { // from class: com.mt.mt166demo.ConnDialog.1
            @Override // android.view.View.OnClickListener
            public void onClick(View v) {
                try {
                    SerialPortItem port = (SerialPortItem) ConnDialog.this.mSpinPorts.getSelectedItem();
                    String baudRate = (String) ConnDialog.this.mSpinBaudRates.getSelectedItem();
                    Application.cr.MtConnect(port.getFullPath(), Integer.parseInt(baudRate), 0);
                    ConnDialog.this.dismiss();
                    SharedPreferences sp2 = ConnDialog.this.mContext.getSharedPreferences("conn_pref", 0);
                    SharedPreferences.Editor editor = sp2.edit();
                    editor.putString("PortName", port.getName());
                    editor.putInt("BaudRate", Integer.parseInt(baudRate));
                    editor.apply();
                    if (ConnDialog.this.mCloseListener != null) {
                        ConnDialog.this.mCloseListener.onClose(true);
                    }
                } catch (Exception e) {
                    Application.showError(e.getMessage(), ConnDialog.this.mContext);
                }
            }
        });
        Button CancelButton = (Button) findViewById(R.id.btnCancel);
        CancelButton.setOnClickListener(new View.OnClickListener() { // from class: com.mt.mt166demo.ConnDialog.2
            @Override // android.view.View.OnClickListener
            public void onClick(View v) {
                try {
                    ConnDialog.this.dismiss();
                    if (ConnDialog.this.mCloseListener != null) {
                        ConnDialog.this.mCloseListener.onClose(false);
                    }
                } catch (Exception e) {
                    Application.showError(e.getMessage(), ConnDialog.this.mContext);
                }
            }
        });
    }
}

File: app/src/main/java/com/mt/mt166demo/ConnDialog.kt
package com.mt.mt166demo

import android.app.Dialog
import android.content.Context
import android.os.Bundle
import android.view.View
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.Spinner
import android.widget.SpinnerAdapter
import java.util.Vector
import androidx.core.content.edit


class ConnDialog(var mContext: Context) : Dialog(mContext) {
    var mCloseListener: OnCloseListener? = null
    var mSpinBaudRates: Spinner? = null
    var mSpinPorts: Spinner? = null

    fun interface OnCloseListener {
        fun onClose(z: Boolean)
    }

    fun setCloseListener(listener: OnCloseListener?) {
        this.mCloseListener = listener
    }

    // android.app.Dialog
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_conn)
        setTitle(MainActivity.Companion.instance!!.getString(R.string.deviceconnect))
        setCancelable(false)
        this.mSpinPorts = findViewById<View?>(R.id.spinPorts) as Spinner
        this.mSpinBaudRates = findViewById<View?>(R.id.spinBaudRates) as Spinner
        val finder = SerialPortFinder()
        val ports = finder.allPorts
        if (ports.isNotEmpty()) {
            val adapter1 = ArrayAdapter<SerialPortItem?>(
                this.mContext,
                R.layout.support_simple_spinner_dropdown_item,
                ports
            )
            this.mSpinPorts!!.adapter = adapter1 as SpinnerAdapter
        } else {
            val ports2 = Vector<SerialPortItem?>()
            ports2.add(SerialPortItem("COM0", "/dev/ttyS0"))
            ports2.add(SerialPortItem("COM1", "/dev/ttyS1"))
            ports2.add(SerialPortItem("COM2", "/dev/ttyS2"))
            ports2.add(SerialPortItem("COM3", "/dev/ttyS3"))
            ports2.add(SerialPortItem("COM4", "/dev/ttyS4"))
            val adapter12 = ArrayAdapter<SerialPortItem?>(
                this.mContext,
                R.layout.support_simple_spinner_dropdown_item,
                ports2
            )
            this.mSpinPorts!!.adapter = adapter12 as SpinnerAdapter
        }
        val baudRates = Vector<String?>()
        baudRates.add("9600")
        baudRates.add("19200")
        baudRates.add("38400")
        baudRates.add("57600")
        baudRates.add("115200")
        val adapter2 = ArrayAdapter<String?>(
            this.mContext,
            R.layout.support_simple_spinner_dropdown_item,
            baudRates
        )
        this.mSpinBaudRates!!.adapter = adapter2 as SpinnerAdapter
        val sp = this.mContext.getSharedPreferences("conn_pref", 0)
        val lastPortName: String = sp.getString("PortName", "")!!
        var i = 0
        while (true) {
            if (i >= this.mSpinPorts!!.count) {
                break
            }
            if (!(this.mSpinPorts!!.getItemAtPosition(i) as SerialPortItem).name
                    .equals(lastPortName, ignoreCase = true)
            ) {
                i++
            } else {
                this.mSpinPorts!!.setSelection(i)
                break
            }
        }
        var i2 = 0
        while (true) {
            if (i2 >= this.mSpinBaudRates!!.count) {
                break
            }
            if ((this.mSpinBaudRates!!.getItemAtPosition(i2) as String?)!!.toInt() != 9600) {
                i2++
            } else {
                this.mSpinBaudRates!!.setSelection(i2)
                break
            }
        }
        val okButton = findViewById<View?>(R.id.btnOK) as Button
        okButton.setOnClickListener {
            try {
                val port = this@ConnDialog.mSpinPorts!!.selectedItem as SerialPortItem
                val baudRate = this@ConnDialog.mSpinBaudRates!!.selectedItem as String
                Application.cr.MtConnect(port.fullPath, baudRate.toInt(), 0)
                this@ConnDialog.dismiss()
                val sp2 = this@ConnDialog.mContext.getSharedPreferences("conn_pref", 0)
                sp2.edit {
                    putString("PortName", port.name)
                    putInt("BaudRate", baudRate.toInt())
                }
                if (this@ConnDialog.mCloseListener != null) {
                    this@ConnDialog.mCloseListener!!.onClose(true)
                }
            } catch (e: Exception) {
                Application.showError(e.message, this@ConnDialog.mContext)
            }
        }
        val cancelButton = findViewById<View?>(R.id.btnCancel) as Button
        cancelButton.setOnClickListener {
            try {
                this@ConnDialog.dismiss()
                if (this@ConnDialog.mCloseListener != null) {
                    this@ConnDialog.mCloseListener!!.onClose(false)
                }
            } catch (e: Exception) {
                Application.showError(e.message, this@ConnDialog.mContext)
            }
        }
    }
}
